profile/main.go

```diff
@@ -425,13 +454,68 @@ func handler(request events.APIGatewayProxyRequest) (events.APIGatewayProxyRespo
 			continue
 		}
 
-		getdata(client, ctx, userId, userUrl, chaincode)
+		status := getdata(client, ctx, userId, userUrl, chaincode)
+		if status == "skippedSameLastPendingDiff" {
+			profilesSkipped.SameAsLastPendingDiff = append(profilesSkipped.SameAsLastPendingDiff, userId)
+		} else if status == "skippedCurrentUserDataSameAsDiff" {
+			profilesSkipped.CurrentUserDataSameAsDiff = append(profilesSkipped.CurrentUserDataSameAsDiff, userId)
+		} else if status == "skippedSameLastRejectedDiff" {
+			profilesSkipped.SameAsLastRejectedDiff = append(profilesSkipped.SameAsLastRejectedDiff, userId)
+		} else if status == "profileSkippedDueToErrorInGettingProfileData" {
+			profilesSkipped.ErrorInGettingProfileData = append(profilesSkipped.ErrorInGettingProfileData, userId)
+		} else if status == "profileSkippedDueToUnAuthenticatedAccessToProfileData" {
+			profilesSkipped.UnAuthenticatedAccessToProfileData = append(profilesSkipped.UnAuthenticatedAccessToProfileData, userId)
+		} else {
+			profileDiffsStored = append(profileDiffsStored, userId)
+		}
+	}
+
+	var report = map[string]interface{}{
+		"TotalProfilesChecked": totalProfilesChecked,
+		"Stored": map[string]interface{}{
+			"count":   len(profileDiffsStored),
+			"userIds": profileDiffsStored,
+		},
+		"Skipped": map[string]interface{}{
+			"CurrentUserDataSameAsDiff": map[string]interface{}{
+				"count":   len(profilesSkipped.CurrentUserDataSameAsDiff),
+				"userIds": profilesSkipped.CurrentUserDataSameAsDiff,
+			},
+			"SameAsLastRejectedDiff": map[string]interface{}{
+				"count":   len(profilesSkipped.SameAsLastRejectedDiff),
+				"userIds": profilesSkipped.SameAsLastRejectedDiff,
+			},
+			"NoProfileURLCount": map[string]interface{}{
+				"count":   len(profilesSkipped.ProfileURL),
+				"userIds": profilesSkipped.ProfileURL,
+			},
+			"UnauthenticatedAccessToProfileData": map[string]interface{}{
+				"count":   len(profilesSkipped.UnAuthenticatedAccessToProfileData),
+				"userIds": profilesSkipped.UnAuthenticatedAccessToProfileData,
+			},
+			"ErrorInGettingProfileData": map[string]interface{}{
+				"count":   len(profilesSkipped.ErrorInGettingProfileData),
+				"userIds": profilesSkipped.ErrorInGettingProfileData,
+			},
+			"ServiceDown": map[string]interface{}{
+				"count":   len(profilesSkipped.ServiceDown),
+				"userIds": profilesSkipped.ServiceDown,
+			},
+			"SameAsLastPendingDiff": map[string]interface{}{
+				"count":   len(profilesSkipped.SameAsLastPendingDiff),
+				"userIds": profilesSkipped.SameAsLastPendingDiff,
+			},
+		},
```

Comment:
Can we make this a function?

Reply:
yes made a new function to generate report